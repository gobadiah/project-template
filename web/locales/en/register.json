{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
