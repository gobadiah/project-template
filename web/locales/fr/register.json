{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
