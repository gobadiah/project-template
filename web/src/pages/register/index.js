import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
