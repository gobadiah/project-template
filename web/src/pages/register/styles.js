import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
