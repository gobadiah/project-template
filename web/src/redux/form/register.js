import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
