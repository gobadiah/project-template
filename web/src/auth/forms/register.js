import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
