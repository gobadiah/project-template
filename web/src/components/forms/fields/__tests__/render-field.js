import 'jsdom-global/register';

import React from 'react';
import { mount, shallow } from 'enzyme';

import RenderField from '../render-field';

const props = {
  input: {
    name: 'example field',
  },
  meta: {
    error: ['This is one error', 'this is another'],
    touched: true,
    warning: ['This is one warning', 'yet another'],
  },
  label: 'Some example field',
};

const context = {
  i18n: {},
  t: jest.fn(),
  user: undefined,
};

describe('RenderField', () => {
  it('should match snapshot', () => {
    const wrapper = mount(<RenderField {...props} />, { context });
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with column=true', () => {
    const wrapper = mount(<RenderField {...props} column />, { context });
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain an input when type is text (as default)', () => {
    const wrapper = shallow(<RenderField {...props} />, { context });
    expect(wrapper.find('input').exists()).toBeTruthy();
  });

  it('should contain a textarea when type is textarea', () => {
    const wrapper = shallow(<RenderField {...props} type='textarea' />, { context });
    expect(wrapper.find('textarea').exists()).toBeTruthy();
  });
});
