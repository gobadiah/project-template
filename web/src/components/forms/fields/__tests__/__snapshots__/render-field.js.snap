// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderField should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<RenderField
  cols={70}
  column={false}
  input={
    Object {
      "name": "example field",
    }
  }
  label="Some example field"
  meta={
    Object {
      "error": Array [
        "This is one error",
        "this is another",
      ],
      "touched": true,
      "warning": Array [
        "This is one warning",
        "yet another",
      ],
    }
  }
  rows={7}
  type="text"
>
  <Styled(div)>
    <div
      className="emotion-0 emotion-1"
    >
      <label
        htmlFor="example field"
        key="label"
      />
      <input
        key="component"
        name="example field"
        type="text"
      />
      <Error
        error={
          Array [
            "This is one error",
            "this is another",
          ]
        }
        key="error"
      >
        <div
          key="0"
        />
        <div
          key="1"
        />
      </Error>
      <Warning
        key="warning"
        warning={
          Array [
            "This is one warning",
            "yet another",
          ]
        }
      >
        <div
          key="0"
        />
        <div
          key="1"
        />
      </Warning>
    </div>
  </Styled(div)>
</RenderField>
`;

exports[`RenderField should match snapshot with column=true 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<RenderField
  cols={70}
  column={true}
  input={
    Object {
      "name": "example field",
    }
  }
  label="Some example field"
  meta={
    Object {
      "error": Array [
        "This is one error",
        "this is another",
      ],
      "touched": true,
      "warning": Array [
        "This is one warning",
        "yet another",
      ],
    }
  }
  rows={7}
  type="text"
>
  <Styled(div)>
    <div
      className="emotion-0 emotion-1"
    >
      <label
        htmlFor="example field"
        key="label"
      />
      <input
        key="component"
        name="example field"
        type="text"
      />
      <Error
        error={
          Array [
            "This is one error",
            "this is another",
          ]
        }
        key="error"
      >
        <div
          key="0"
        />
        <div
          key="1"
        />
      </Error>
      <Warning
        key="warning"
        warning={
          Array [
            "This is one warning",
            "yet another",
          ]
        }
      >
        <div
          key="0"
        />
        <div
          key="1"
        />
      </Warning>
    </div>
  </Styled(div)>
</RenderField>
`;
